plugins/modules/volume.py

```diff
@@ -175,14 +186,99 @@ def poll_func() -> bool:
                 "failed to wait for volume status: timeout period expired"
             )
 
-    def _wait_for_volume_active(self) -> None:
+    def _wait_for_volume_active(self, volume: Volume) -> None:
         self._wait_for_volume_status(
-            self._volume, {"active"}, self._timeout_ctx.seconds_remaining
+            volume, {"active"}, self._timeout_ctx.seconds_remaining
+        )
+
+    def _clone_volume(self) -> Volume:
+        params = self.module.params
+
+        source_id = params.get("source_volume_id")
+        source_volume = Volume(self.client, source_id)
+        source_volume._api_get()  # Force lazy-loading
+
+        # Check if source volume was found
+        if source_volume is None:
+            return self.fail(
+                msg="Volume with is {0} could not be found.".format(source_id)
+            )
+
+        # If regions don't match up, it is an invalid clone operation
+        if params.get("region") != source_volume.region.id:
+            return self.fail(
+                msg="Specified region does not match source volume region."
+            )
+
+        resize = False
+
+        if params.get("size") != source_volume.size:
+            resize = True
+
+        # Perform the clone operation
+        vol = request_retry(
+            lambda: self.client.post(
+                "/volumes/{}/clone".format(source_id),
+                data={"label": params.get("label")},
+            )
         )
 
+        cloned_volume = Volume(self.client, vol.get("id"))
+        cloned_volume._api_get()  # Force lazy-loading
+
+        self.register_action(
+            "Cloned volume with source id {0}".format(source_id)
+        )
+        self._wait_for_volume_active(cloned_volume)
+
+        # Resize if necessary
+        if resize:
```

Comment:
The `resize` flag seems only be used here. Should we do 
`if params.get("size") != source_volume.size` directly instead of storing the boolean value?